src/query/v2/interpreter.hpp

```diff
@@ -180,6 +185,9 @@ struct InterpreterContext {
 
   const InterpreterConfig config;
 
+  io::Io<io::local_transport::LocalTransport> io;
```

Comment:
We have to figure out an abstraction for `io::Io` to make it possible to construct an interpreter context with a simulator transport without templatizing it. Maybe some kind of client factor or something. It is not in the scope of this PR, just raising awareness.

Reply:
Will leave TODO